lib/mayaUsd/ufe/private/Utils.cpp

```diff
@@ -191,6 +216,53 @@ void applyCommandRestriction(const UsdPrim& prim, const std::string& commandName
     }
 }
 
+bool isAttributeEditAllowed(const PXR_NS::UsdAttribute& attr)
+{
+    // get the property spec in the edit target's layer
+    const auto& prim = attr.GetPrim();
+    const auto& stage = prim.GetStage();
+    const auto& editTarget = stage->GetEditTarget();
+
+    // get the index to edit target layer
+    const auto targetLayerIndex = findLayerIndex(attr, editTarget.GetLayer());
+
+    // get the strength-ordered ( strong-to-weak order ) list of property specs that provide
+    // opinions for this property.
+    const auto& propertyStack = attr.GetPropertyStack();
+
+    // HS March 15th,2021
+    // TODO: This code works as long as existing opinions are in the stage’s local LayerStack.
+    // Relying on the "expanded primIndex" would be a better way to detect the opinions across
+    // composition arc(s). Some more details can be found:
+    // https://groups.google.com/g/usd-interest/c/xTxFYQA_bRs/m/lX_WqNLoBAAJ
+    SdfLayerHandle strongLayer;
+    for (const auto& spec : propertyStack) {
+
+        // skip if the edit target layer is stronger than the propSpec layer
+        const auto propSpecLayerIndex = findLayerIndex(attr, spec->GetLayer());
+        if (targetLayerIndex <= propSpecLayerIndex) {
+            continue;
+        }
+
+        if (spec) {
+            strongLayer = spec->GetLayer();
+            break;
+        }
+    }
+
+    if (strongLayer) {
+        std::string err = TfStringPrintf(
+            "Cannot edit [%s] attribute because there is a stronger opinion in [%s].",
+            attr.GetBaseName().GetText(),
+            strongLayer->GetDisplayName().c_str());
+
+        MGlobal::displayError(err.c_str());
```

Comment:
I originally intended to just "throw std::runtime_error" or "throw" but that won't work considering the very long
error message that we would get when the exceptions are thrown.

`# Error: RuntimeError: file <string> line 2: RuntimeError: file C:\Users\sabrih\maya\builds\master\maya\build\RelWithDebInfo\runTime\Python\lib\site-packages\maya\internal\ufeSupport\ufeCmdWrapper.py line 32: Cannot edit [height] attribute because there is a stronger opinion in [anonymousLayer4].`

Instead this function returns a bool and simply shows an error message using MGlobal::displayError

Reply:
@ppt-adsk Good point. addressed in fe8131c1c837ebf7bf5405e7c594df7b5b7b1db4